recipes-azure-iot/iot-hub-device-update/iot-hub-device-update/swupdate_handler_v2_grub.sh

```diff
@@ -479,13 +485,112 @@ function is_installed() {
     fi
 }
 
+#
+# A helper function that compares two software versions.
+#
+# version number schema:
+# - based on Major.Minor.Patch.Buildnumber
+# - based only on digits
+#
+# If version1 newer than version2           -> return 1
+# If version1 older than version2           -> return 2
+# If version1 equal version2                -> return 3
+# Illegal format in version1 or version2    -> return 0
+#
+# Usage: version_comp $version1 $version2
+#
+# shellcheck disable=SC2034
+function version_comp () {
+
+    local regex='^([0-9]+)\.([0-9]+)\.([0-9]+)\.([0-9]+)$'
```

Comment:
why don't we exclude leading zeros here?

Reply:
removed sed and changed regex